src/history.rs

```diff
@@ -1040,7 +855,12 @@ impl HistoryImpl {
         } else {
             // If we have not loaded old items, don't actually load them (which may be expensive); just
             // stat the file and see if it exists and is nonempty.
-            let Some(where_) = history_filename(&self.name, L!("")) else {
+
+            if self.name.is_empty() {
+                return true;
+            }
```

Comment:
this is purely an early exit for the `set fish_history ''` case, right?
To avoid the `Box<&str>` allocation done by `std::io::Error::new()`.

This is a bit awkward because there are other callers of `self.history_file_path()` such as `save_internal_via_appending()`
that don't check for the empty case; it seems weird to be inconsistent.

Indeed this change logs `Rewriting history failed:` in this case.
I think we should not; it's not an error; we should silently not write the history (but still keep it in memory).

Can we change `self.history_file_path()` to return `std::io::Result<Option<WString>>`
and return `Ok(None)` if the name is empty meaning "everything is fine but don't write the history to disk".